stripe/src/main/java/com/stripe/android/CustomerSession.java

```diff
@@ -264,8 +281,12 @@ public void deleteCustomerSource(
         mContext = context.getApplicationContext();
         Map<String, Object> arguments = new HashMap<>();
         arguments.put(KEY_SOURCE, sourceId);
-        mSourceRetrievalListener = listener;
-        mEphemeralKeyManager.retrieveEphemeralKey(ACTION_DELETE_SOURCE, arguments);
+
+        final String operationId = UUID.randomUUID().toString();
+        if (listener != null) {
+            mSourceRetrievalListenerRefs.put(operationId, new WeakReference<>(listener));
+        }
+        mEphemeralKeyManager.retrieveEphemeralKey(operationId, ACTION_DELETE_SOURCE, arguments);
```

Comment:
Is the `operationId` needed if the listener is null?

Reply:
see below